internal/fsutil.go

```diff
@@ -72,6 +73,50 @@ func createTestExtImg(ctx context.Context, t *testing.T, extName string, testFil
 	return imgFile.Name()
 }
 
+// BlockDeviceFile represents a block device
+type BlockDeviceFile struct {
+	Subpath string
+	UID     int
+	GID     int
+	Dev     int // major number
+}
+
+// CreateBlockDevice creates a block device, or block special file for testing
+func CreateBlockDevice(ctx context.Context, t *testing.T, fsType string, testFile BlockDeviceFile) string {
+	t.Helper()
+
+	switch fsType {
+	case "ext4":
+		return createTestBlockDevice(ctx, t, fsType, testFile)
+	default:
+		require.FailNowf(t, "unsupported fs type %q", fsType)
+		return ""
+	}
+}
+
+func createTestBlockDevice(ctx context.Context, t *testing.T, extName string, testFile BlockDeviceFile) string {
+	t.Helper()
+
+	deviceFile := filepath.Join("/tmp/blockExample", testFile.Subpath)
```

Comment:
Can't you use os.TempDir or ioutil.TempDir?

Reply:
No. This will introduce problem due to parallel testing.